RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSCProducer.h

```diff
@@ -0,0 +1,41 @@
+#ifndef LowPtGsfElectronSCProducer_h
+#define LowPtGsfElectronSCProducer_h
+
+#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/GsfPFRecTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/GsfPFRecTrackFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFClusterFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFTrajectoryPoint.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+class LowPtGsfElectronSCProducer : public edm::stream::EDProducer<> {
+  
+ public:
+  
+  explicit LowPtGsfElectronSCProducer( const edm::ParameterSet& );
+
+  ~LowPtGsfElectronSCProducer() override;
+
+  void produce( edm::Event&, const edm::EventSetup& ) override;
+
+ private:
+
+  typedef edm::Ptr<reco::PFCluster> PFClusterPtr;
+  typedef reco::PFClusterCollection PFClusters;
+  typedef reco::GsfPFRecTrackCollection GsfPFRecTracks;
+  typedef reco::SuperClusterCollection SuperClusters;
+  typedef reco::PFTrajectoryPoint::LayerType LayerType;
+    
```

Comment:
please dont make unnecessary typedefs, its a net loss to efficiency. Sure you are faster typing but now I had to go an look up what SuperClusters was so it slows down anybody reading the code. 

also typedef is out of favour, an alias "using X = Y;" is now the preferred way due to how it interacts with templates.

Reply:
Removed typedefs